runtime/decimal.c

```diff
@@ -47,5 +47,10 @@ TaggedPtrPanicCode _bal_decimal_add(TaggedPtr tp1, TaggedPtr tp2) {
     return result;    
```

Comment:
This will occur in all the operations. So let's factor out a function:

```
static inline TaggedPtrPanicCode finish(decQuad *dq, decContext *cx) {
    TaggedPtrPanicCode result;
    result.panicCode = statusPanicCode(cx->status);
    if (result.panicCode == 0) {
        result.ptr = createDecimal(dq);
    }
    return result;  
}
```

We we will also need to deal with negative zero (which I think can only happen with underflow). So let's inline staticPanicCode into finish above.

Reply:
@jclark, we need to add this method to the `_bal_decimal_const` right? because in that method also we have to deal with errors like underflow/overflow.

Reply:
No: these are constants generated by the compiler. The compiler needs to generate only valid constants.

Reply:
Fixed